src/main/java/org/sciborgs1155/robot/drivetrain/SparkDiffDrive.java

```diff
@@ -0,0 +1,188 @@
+    // This leads the 'rearRightMotor', so we do not have to set both
+    frontLeftMotor.setVoltage(clampVoltage(voltage).in(Volts));
+
+    return voltage;
+  }
+
+  /**
+   * Total displacement of the left side of the drivetrain (averages the velocities of both motors
+   * for robustness): i.e. both motors going opposite directions
+   */
+  @Override
+  public Measure<Distance> getLeftDisplacement() {
+    return Meters.of((frontLeftEncoder.getPosition() + rearLeftEncoder.getPosition()) / 2);
+  }
+
+  /**
+   * The current velocity of the left side of the drivetrain (averages the velocities of both motors
+   * for robustness): i.e. both motors going opposite directions
+   */
+  @Override
+  public Measure<Velocity<Distance>> getLeftVelocity() {
+    return MetersPerSecond.of((frontLeftEncoder.getVelocity() + rearLeftEncoder.getVelocity()) / 2);
+  }
+
+  @Override
+  public void resetLeftEncoder() {
+    frontLeftEncoder.setPosition(0);
+    rearLeftEncoder.setPosition(0);
+  }
+
+  @Override
+  public Measure<Voltage> setRightVoltage(Measure<Voltage> voltage) {
+    // This leads the 'rearRightMotor', so we do not have to set both
+    frontRightMotor.setVoltage(clampVoltage(voltage).in(Volts));
+
+    return voltage;
+  }
+
+  /**
+   * Total displacement of the right side of the drivetrain (averages the velocities of both motors
+   * for robustness): i.e. both motors going opposite directions
+   */
+  @Override
+  public Measure<Distance> getRightDisplacement() {
+    return Meters.of((frontRightEncoder.getPosition() + rearRightEncoder.getPosition()) / 2);
+  }
+
+  /**
+   * The current velocity of the right side of the drivetrain (averages the velocities of both
+   * motors for robustness): i.e. both motors going opposite directions
+   */
+  @Override
+  public Measure<Velocity<Distance>> getRightVelocity() {
+    return MetersPerSecond.of(
+        (frontRightEncoder.getVelocity() + rearRightEncoder.getVelocity()) / 2);
+  }
+
+  @Override
+  public Measure<Velocity<Angle>> getAngularVelocity() {
+    return DegreesPerSecond.of(previousDisplacements[3]);
+  }
+
+  @Override
+  public void resetRightEncoder() {
+    frontRightEncoder.setPosition(0);
+    rearRightEncoder.setPosition(0);
+  }
+
+  /**
+   * @param motorIDs : [Front Left, Rear Left, Front Right, Rear Right]
+   */
+  public SparkDiffDrive(int[] motorIDs) {
+    this.frontLeftMotor = new CANSparkMax(motorIDs[0], MotorType.kBrushless);
+    this.rearLeftMotor = new CANSparkMax(motorIDs[1], MotorType.kBrushless);
+    this.frontRightMotor = new CANSparkMax(motorIDs[2], MotorType.kBrushless);
+    this.rearRightMotor = new CANSparkMax(motorIDs[3], MotorType.kBrushless);
+
+    this.frontLeftEncoder = frontLeftMotor.getEncoder();
+    this.rearLeftEncoder = rearLeftMotor.getEncoder();
+    this.frontRightEncoder = frontRightMotor.getEncoder();
+    this.rearRightEncoder = rearRightMotor.getEncoder();
+
+    this.frontLeftMotor.restoreFactoryDefaults();
```

Comment:
For these configs, you should use SparkUtils' `check` method to ensure that you know if they fail or not.